Formula/opencv.rb

```diff
@@ -92,43 +96,39 @@ def install
       -DWITH_VTK=ON
       -DBUILD_opencv_python2=OFF
       -DBUILD_opencv_python3=ON
-      -DPYTHON3_EXECUTABLE=#{Formula["python@3.9"].opt_bin}/python3
+      -DPYTHON3_EXECUTABLE=#{Formula["python@3.10"].opt_bin}/python3
     ]
 
     # Disable precompiled headers and force opencv to use brewed libraries on Linux
     if OS.linux?
-      args << "-DENABLE_PRECOMPILED_HEADERS=OFF"
-      args << "-DJPEG_LIBRARY=#{Formula["libjpeg"].opt_lib}/libjpeg.so"
-      args << "-DOpenBLAS_LIB=#{Formula["openblas"].opt_lib}/libopenblas.so"
-      args << "-DOPENEXR_ILMIMF_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmImf.so"
-      args << "-DOPENEXR_ILMTHREAD_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmThread.so"
-      args << "-DPNG_LIBRARY=#{Formula["libpng"].opt_lib}/libpng.so"
-      args << "-DPROTOBUF_LIBRARY=#{Formula["protobuf"].opt_lib}/libprotobuf.so"
-      args << "-DTIFF_LIBRARY=#{Formula["libtiff"].opt_lib}/libtiff.so"
-      args << "-DWITH_V4L=OFF"
-      args << "-DZLIB_LIBRARY=#{Formula["zlib"].opt_lib}/libz.so"
+      args += %W[
+        -DENABLE_PRECOMPILED_HEADERS=OFF
+        -DJPEG_LIBRARY=#{Formula["jpeg-turbo"].opt_lib}/libjpeg.so
+        -DOpenBLAS_LIB=#{Formula["openblas"].opt_lib}/libopenblas.so
+        -DOPENEXR_ILMIMF_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmImf.so
+        -DOPENEXR_ILMTHREAD_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmThread.so
+        -DPNG_LIBRARY=#{Formula["libpng"].opt_lib}/libpng.so
+        -DPROTOBUF_LIBRARY=#{Formula["protobuf"].opt_lib}/libprotobuf.so
+        -DTIFF_LIBRARY=#{Formula["libtiff"].opt_lib}/libtiff.so
+        -DWITH_V4L=OFF
+        -DZLIB_LIBRARY=#{Formula["zlib"].opt_lib}/libz.so
+      ]
     end
 
-    if Hardware::CPU.intel?
-      args << "-DENABLE_AVX=OFF" << "-DENABLE_AVX2=OFF"
-      args << "-DENABLE_SSE41=OFF" << "-DENABLE_SSE42=OFF" unless MacOS.version.requires_sse42?
+    ENV.runtime_cpu_detection
+    if Hardware::CPU.intel? && build.bottle?
+      args << "-DCPU_BASELINE_REQUIRE=#{MacOS.version.requires_sse42? ? "SSE4_2" : "SSSE3"}"
     end
```

Comment:
May need someone to double check this. Ref: https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options